nmostesting/suites/IS1101Test.py

```diff
@@ -0,0 +1,3813 @@
+        Verify for inputs supporting EDID that the version and the effective EDID change when applying constraints
+        """
+        if len(self.flow_format_audio) == 0:
+            return test.UNCLEAR("There is no audio format")
+
+        for sender_id in self.flow_format_audio:
+            valid, response = TestHelper.do_request(
+                "GET", self.compat_url + "senders/" + sender_id + "/inputs/"
+            )
+            if not valid:
+                return test.FAIL(
+                    "Unexpected response from the streamcompatibility API: {}".format(
+                        response
+                    )
+                )
+            if response.status_code != 200:
+                return test.FAIL(
+                    "The sender {} inputs streamcompatibility request has failed: {}".format(
+                        sender_id, response
+                    )
+                )
+            inputs = []
+            try:
+                for input_id in response.json():
+                    if (
+                        input_id in self.edid_connected_inputs
+                        and input_id in self.base_edid_inputs
+                    ):
+                        inputs.append(input_id)
+                    else:
+                        print(
+                            "Inputs {} are not connected or does'nt support base Edid".format(
+                                input_id
+                            )
+                        )
+                        break
+            except json.JSONDecodeError:
+                return test.FAIL("Non-JSON response returned from Node API")
+            except KeyError as e:
+                return test.FAIL("Unable to find expected key: {}".format(e))
+
+            if len(inputs) == 0:
+                return test.UNCLEAR("No input supports changing the base EDID")
+            for input_id in inputs:
+                valid, response = TestHelper.do_request(
+                    "GET", self.compat_url + "inputs/" + input_id + "/properties/"
+                )
+                if not valid:
+                    return test.FAIL(
+                        "Unexpected response from the streamcompatibility API: {}".format(
+                            response
+                        )
+                    )
+                if response.status_code != 200:
+                    return test.FAIL(
+                        "The input {} properties streamcompatibility request has failed: {}".format(
+                            input_id, response
+                        )
+                    )
+                try:
+                    version = response.json()["version"]
+                except json.JSONDecodeError:
+                    return test.FAIL("Non-JSON response returned from Node API")
+                except KeyError as e:
+                    return test.FAIL("Unable to find expected key: {}".format(e))
+                self.version[input_id] = version
```

Comment:
This seems to be a repeated pattern through this test. Perhaps factoring out a helper function wold allow this test to be more compact?